zebra-state/src/service/tests.rs

```diff
@@ -183,3 +187,172 @@ fn state_behaves_when_blocks_are_committed_out_of_order() -> Result<()> {
 
     Ok(())
 }
+
+/*
+#[test]
```

Comment:
I haven't reviewed the tests in detail yet.

Here are some changes I'd like to see first:
1. Do the `legacy_chain_check` and proptest strategy refactors

2. Delete any redundant tests

3. Test Coverage: Make sure every rule is covered - see the latest list of test cases in the ticket

4. Explain tests: Can you add a short doc comment to each test, explaining which rule it tests?